### PR TITLE
add support for checking required/optional EasyBuild dependencies via 'eb --check-eb-deps'

### PR DESCRIPTION
**Pull request by boegel**
Example output when using Python 3 and having `rich` installed (misaligned lines in tables is only due to copy-pasting)

```
$ eb --check-eb-deps

     Required dependencies
┏━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ name         ┃ version      ┃
┡━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━┩
│ Python       │ ✅  3.9.5    │
│ modules tool │ ✅  Lmod 8.3 │
└──────────────┴──────────────┘
                                          Optional dependencies
┏━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ name              ┃ version       ┃ used for                                                          ┃
┡━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ archspec          │ ✅  0.1.2     │ determining name of CPU microarchitecture                         │
│ autopep8          │ ✅  1.5.7     │ auto-formatting for dumped easyconfigs                            │
│ GC3Pie            │ ✅  2.6.8     │ backend for --job                                                 │
│ GitPython         │ ✅  3.1.11    │ GitHub integration + using Git repository as easyconfigs archive  │
│ graphviz-python   │ ❌  not found │ rendering dependency graph with Graphviz: --dep-graph             │
│ keyring           │ ✅  21.4.0    │ storing GitHub token                                              │
│ pbs-python        │ ❌  not found │ using Torque as --job backend                                     │
│ pep8              │ ❌  not found │ fallback for code style checking: --check-style, --check-contrib  │
│ pycodestyle       │ ✅  2.7.0     │ code style checking: --check-style, --check-contrib               │
│ pysvn             │ ❌  not found │ using SVN repository as easyconfigs archive                       │
│ python-graph-core │ ❌  not found │ creating dependency graph: --dep-graph                            │
│ python-graph-dot  │ ❌  not found │ saving dependency graph as dot file: --dep-graph                  │
│ python-hglib      │ ✅  2.6.2     │ using Mercurial repository as easyconfigs archive                 │
│ PyYAML            │ ✅  5.3.1     │ easystack files and .yeb easyconfig format                        │
│ requests          │ ✅  2.24.0    │ fallback library for downloading files                            │
│ Rich              │ ✅  10.4.0    │ eb command rich terminal output                                   │
│ setuptools        │ ✅  56.0.0    │ obtaining information on Python packages via pkg_resources module │
└───────────────────┴───────────────┴───────────────────────────────────────────────────────────────────┘
        System tools
┏━━━━━━━━━┳━━━━━━━━━━━━━━━━┓
┃ name    ┃ version        ┃
┡━━━━━━━━━╇━━━━━━━━━━━━━━━━┩
│ 7z      │ ❌  not found  │
│ bunzip2 │ ✅  1.0.6      │
│ dpkg    │ ❌  not found  │
│ gunzip  │ ✅  321.100.11 │
│ make    │ ✅  3.81       │
│ patch   │ ✅  2.5.8      │
│ rpm     │ ❌  not found  │
│ sed     │ 🟡  version?!  │
│ Slurm   │ ❌  not found  │
│ tar     │ ✅  3.3.2      │
│ unxz    │ ✅  5.2.5      │
│ unzip   │ ✅  6.00       │
└─────────┴────────────────┘
```

### CODE REVIEW

File: easybuild/tools/output.py [hunk 1]
```diff
@@ -82,3 +85,71 @@ def create_progress_bar():
...
+    """Print overview of checks that were made."""
+
+    col_titles = checks_data.pop('col_titles', ('name', 'info', 'description'))
+
+    col2_label = col_titles[1]
+
+    if HAVE_RICH:
+        console = Console()
+        # don't use console.print, which causes SyntaxError in Python 2
+        console_print = getattr(console, 'print')
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/tools/output.py:100]: Erm, I disagree, see the comment above. Bad dog!

File: easybuild/tools/systemtools.py [hunk 2]
```diff
@@ -152,6 +160,56 @@
...
+}
+
+SYSTEM_TOOL_CMDS = {
+    'Slurm': 'sbatch',
+}
+
+OPT_DEPS = {
+    'archspec': "determining name of CPU microarchitecture",
+    'autopep8': "auto-formatting for dumped easyconfigs",
+    'GC3Pie': "backend for --job",
```
Comment (akesandgren) [inline easybuild/tools/systemtools.py:185]: We used this through EB installation when we used it... and we do have it in EB...

Comment (boegel) [reply easybuild/tools/systemtools.py]: Doesn't matter much, the point is to check all for optional dependencies whether they're available or not, it doesn't matter how they're installed...

**Review by akesandgren — APPROVED** [review]
LGTM, don't have rich installed but works nicely without it.

**Pull request merged**